### ISSUE TITLE
Unable to replace/verify website proof due to vague error

### ISSUE THREAD

**Issue by breadtk** [keybase/keybase-issues#1459]
Hello,

I recently changed a few bits on my server and it appears to have broken your verification mechanism for websites. The domain in question is surkatty.org. When I attempt to replace my proof, I receive the following error: `could not query surkatty.org. Are you running a site there?`.

It isn't clear to me from the error whether the issue is on my end (the server being down, unresolveable, etc.) or on your end (unable to process page, etc.). My assumption is it's on your end since it appears my site is [publically accessible](http://www.downforeveryoneorjustme.com/surkatty.org).

Please let me know what additional things I can provide to help track down this issue.

**Comment by malgorithms** [comment]
Hi @breadtk - thanks for the report.

It appears the core http/https libraries in Node.js do not like the funky headers you're serving up on your site. I don't know which the problem is, but here's what I see when I do a curl:

```
curl -I https://surkatty.org
HTTP/1.1 200 OK
Server: nginx
Date: Mon, 16 Mar 2015 19:29:21 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 61035
Last-Modified: Thu, 12 Feb 2015 22:11:47 GMT
Connection: keep-alive
Vary: Accept-Encoding
ETag: "54dd2523-ee6b"
Expires: Thu, 16 Apr 2015 19:29:21 GMT
Cache-Control: max-age=2678400
Cache-Control: public
Strict-Transport-Security: max-age=42424242; includeSubDomains
X-Frame-Options: DENY
X-Crack-me: 6e55453070555a36596c393371337068724a3931715549764d4635776f323069713253304c32742f717731364d5253734b6d5730463239574946413043475230424b5a7420 
X-Long-header: 
. . . . . . . . . . . . . . . . . . . ,,_ . . . . . . . . ,-'`-,
. . . . . . . . . . . . . . . . . . . .\,, '``-.__.,.,./ .,., .\
. . . . . . . . . . . . . . . . . . . . .|,.-;`;` . . ,.,., .,.,.,`-, . . . . . . . . . . .
..,,,.,.,._ . . . . . . . . . . . . . . / :o;. . .;o; .['. . . . `'-.,. . . . . . . .
['. . . . . . '`'`'`'`'*-----,.,.,.,._\ . . . -;- . . . . '`-,._ . . . `'-., . . . . .
'``'*----,,,,,.,.,.,.,_. . . . . . . ..\,, . ,-,-,. . . . . ,.,. `'-, . . . .\,. . . .
. . . . . . . . . . . . . . ``'`'*----.,., .`'`-----'`'`'`` . . . . .`-. . . . \` . .
. . . . . . . . . . . . . . . . . . . . . .`'-. . . . . . . . . . . . . . . . . . .,| . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . . ,|. . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
. . . . . . . . . . . . . . . . . . . . . . . |;. . . . . . . . . . . . . . . . .,|. . . .
Accept-Ranges: bytes
```

Deep in node's `https` library, it calls back with an error: `HPE_INVALID_CONSTANT` if I try to visit the site, even with a rudimentary node app:

```
var https = require('https');

var options = {
  hostname: 'surkatty.org',
  method: 'GET'
};

var req = https.request(options, function(res) {
  console.log("" + options.hostname + " res.statusCode=" + res.statusCode);
  res.on('data', function (chunk) {
      console.log('BODY: ' + chunk);
    });
});

req.on('error', function (err) {
  console.log(err);
});

req.end();
```

This outputs the following:

```
{ [Error: Parse Error] bytesParsed: 632, code: 'HPE_INVALID_HEADER_TOKEN' }
```

Since we've got thousands of proofs that work we're not in a position to get the `http` module updated easily, I'm assuming this is something you'll need to change on your end.

**Comment by breadtk** [comment]
I'm sorry, it looks like I failed to include in my original response what it is I actually changed on my end. What I did was instead of relying on my server's HSTS headers to redirect all visitors to use HTTPS, I added a 301 for all HTTP requests towards my domain. Prior to this change the scraper was able to crawl the HTTP version of my site which doesn't have the `X-Long-header` header.

I have no plan to rollback this change on my end. So if you're not going to update the scraper to ignore unknown headers on your end, then I just won't have that specific proof. Thank you for taking a look into it.

**Comment by dtiersch** [comment]
You misunderstood @malgorithms. The error is not in the scraper or within keybase itself. The error is in Node, so it can't be fixed by the keybase developers.

**Comment by malgorithms** [comment]
Yeah, I honestly know too little about http to know if it's an "error" in Node.js or if it's legitimately an invalid header token that you're serving (which browsers ignore, so it appears fine). But as @dtiersch said, I'd consider it out of Keybase's hands. Either Node's `http(s)` modules need to change, or your site's headers do. I would call Keybase an uncomfortable MITM on this.

Sorry I don't have a nicer answer. We're not going to write our own http module on top of `net` to handle a case like this.

As a final point: while you don't get the benefits of a 2-way proof and the auditable signature chain, you can at least host a signed statement on your website stating your keybase username and other identities.

**Comment by breadtk** [comment]
@dtiersch: Understood. I agree with @malgorithms, there isn't much that Keybase can do in this case which is why I decided to close the issue. The fix ought to come from Node, unfortunately they've confirmed in the past that they [won't fix this issue](https://github.com/joyent/node/issues/4394#issuecomment-11189222), because it's a feature. :disappointed_relieved:

@malgorithms: I've done a non-signed equivelant link [here](https://surkatty.org/contact.html) and the keybase proof [here](https://surkatty.org/keybase.txt) in the past. I was just hoping to have it bubble back up to my Keybase profile. Thank you again for looking into this.

On the bright side, I'm sure I broke a whole bunch of other Node crawlers that will inevitably visit my site. I'm sure one or two of them will stumble on this issue and just exclude my site from crawling.. :smile_cat:
